filebeat/prospector/prospector.go

```diff
@@ -105,6 +105,10 @@ func (p *Prospector) Run() {
 				logp.Info("Prospector channel stopped")
 				return
 			case event := <-p.harvesterChan:
+				// Add ttl if cleanOlder is enabled
+				if p.config.CleanOlder != 0 {
+					event.FileState.Ttl = int64(p.config.CleanOlder.Seconds())
```

Comment:
event.FileState only forwarded to spooler and registrar next or also used as is in prospector for cleanup? If shared with prospector we need atomic or mutex operation.